pkg/credentials/kubernetes/gar/service_account_key.go

```diff
@@ -42,7 +43,13 @@ func (p *ServiceAccountKeyProvider) Supports(
 	data map[string][]byte,
 	_ map[string]string,
 ) bool {
-	if credType != credentials.TypeImage || data == nil || data[serviceAccountKeyKey] == nil {
+	if credType != credentials.TypeImage && credType != credentials.TypeHelm ||
+		data == nil ||
+		data[serviceAccountKeyKey] == nil {
+		return false
+	}
+
+	if credType == credentials.TypeHelm && !strings.HasPrefix(repoURL, "oci://") {
 		return false
 	}
```

Comment:
Related to [this comment](https://github.com/akuity/kargo/pull/4637/files#r2440880558), I believe this conditional _must_ be removed. (Normalization of a chart URL would have stripped the `oci://` prefix.)